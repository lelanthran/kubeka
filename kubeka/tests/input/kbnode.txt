# A small test file - the actual jobs in this file will
# never be run, it's for testing the parsing of nodes only.


# Some variables set by the user
[cron]
appname = $(env_read APPNAME)
git_origin = ssh://some-origin/some-repo.git
git_branch = master
build_artifacts =
build_artifacts = $(build_artifacts) bin/program
build_artifacts = $(build_artifacts) info/release-info.txt
target_directory = /var/www/$(appname)
test_string = one
test_string += two
test_string += three
test_array = [four, five, six]
test_array[] += seven
test_empty_array[] += eight
test_empty_string += nine


[job]
ID = Full deployment
MESSAGE = Performing a full deployment
JOBS[] = [ build, upgrade, Done ]

[job]
ID = build
MESSAGE = Starting build process
JOBS[] = [ Check changes, checkout, Build thing, test, package ]

[job]
ID = upgrade
MESSAGE = Performing upgrade
JOBS[] = [ pre-deployment, Upgrade DB, Copy files]

[job]
ID = Check changes
MESSAGE = Checking if sources changed, will continue only if sources changed
EXEC += if [ `git  ls-remote $(git_origin) heads/master | cut -f 1 ` == $(current HASH)]; then
EXEC +=    exit -1;
EXEC += else
EXEC +=    exit 0;

[job]
ID = checkout   # Comments work like this
MESSAGE = A single long line message up to 8kb
ROLLBACK = # Nothing to do if this fails
EXEC = git clone $(git_origin) && git checkout $(git_branch) $(_WORKING_PATH)

[job]
ID = Build thing
MESSAGE = Building $(_JOB) for $(_DEPLOYMENT) with parameter $1
ROLLBACK = # Nothing to do if this fails
EXEC = make $1

[job]
ID = test
MESSAGE = Testing $(_DEPLOYMENT)
ROLLBACK = # Nothing to do if this fails
EXEC = test.sh

[job]
ID = package
MESSAGE = Packaging $(_DEPLOYMENT)
EXEC = tar -zcvf $(_TMPDIR)/$(env_read APPNAME).tar.gz $(build_artifacts)

[job]
ID = pre-deployment
MESSAGE = Deploying to $(target_directory)
WORKING_DIR = $(target_directory)
EXEC = systemctl stop && tar -zxvf $(env_read APPNAME)

[job]
ID = Upgrade DB
MESSAGE = Upgrading database
WORKING_DIR = $(target_directory)
ROLLBACK = $(exec ./rollback.sh)
EXEC = db/upgrade.sh

[job]
ID = Copy files
MESSAGE = Copying files to $(target_directory)
EXEC = cp -Rv $(artifacts) $(target_directory)

[job]
ID = Done
MESSAGE = Setting finished variables
HASH = $(exec git log -1 | head -n 1 | cut -f 1 -d \  )



