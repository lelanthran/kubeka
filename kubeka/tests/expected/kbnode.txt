Node [cron] with parent [(nil)]
   appname: [ $(env_read APPNAME) ]
   test_array: [ four ,five ,six ,seven ]
   build_artifacts: [ $(build_artifacts) info/release-info.txt ]
   test_empty_array: [ eight ]
   test_string: [ one two three ]
   git_branch: [ master ]
   target_directory: [ /var/www/$(appname) ]
   test_empty_string: [  nine ]
   git_origin: [ ssh://some-origin/some-repo.git ]
Node [job] with parent [(nil)]
   JOBS: [ build ,upgrade ,Done ]
   MESSAGE: [ Performing a full deployment ]
   ID: [ Full deployment ]
Node [job] with parent [(nil)]
   JOBS: [ Check changes ,checkout ,Build thing ,test ,package ]
   MESSAGE: [ Starting build process ]
   ID: [ build ]
Node [job] with parent [(nil)]
   JOBS: [ pre-deployment ,Upgrade DB ,Copy files ]
   MESSAGE: [ Performing upgrade ]
   ID: [ upgrade ]
Node [job] with parent [(nil)]
   EXEC: [  if [ `git  ls-remote $(git_origin) heads/master | cut -f 1 ` == $(current HASH)]; then exit -1; else exit 0; ]
   MESSAGE: [ Checking if sources changed, will continue only if sources changed ]
   ID: [ Check changes ]
Node [job] with parent [(nil)]
   ROLLBACK: [  ]
   EXEC: [ git clone $(git_origin) && git checkout $(git_branch) $(_WORKING_PATH) ]
   MESSAGE: [ A single long line message up to 8kb ]
   ID: [ checkout ]
Node [job] with parent [(nil)]
   ROLLBACK: [  ]
   EXEC: [ make $1 ]
   MESSAGE: [ Building $(_JOB) for $(_DEPLOYMENT) with parameter $1 ]
   ID: [ Build thing ]
Node [job] with parent [(nil)]
   ROLLBACK: [  ]
   EXEC: [ test.sh ]
   MESSAGE: [ Testing $(_DEPLOYMENT) ]
   ID: [ test ]
Node [job] with parent [(nil)]
   EXEC: [ tar -zcvf $(_TMPDIR)/$(env_read APPNAME).tar.gz $(build_artifacts) ]
   MESSAGE: [ Packaging $(_DEPLOYMENT) ]
   ID: [ package ]
Node [job] with parent [(nil)]
   WORKING_DIR: [ $(target_directory) ]
   EXEC: [ systemctl stop && tar -zxvf $(env_read APPNAME) ]
   MESSAGE: [ Deploying to $(target_directory) ]
   ID: [ pre-deployment ]
Node [job] with parent [(nil)]
   ROLLBACK: [ $(exec ./rollback.sh) ]
   WORKING_DIR: [ $(target_directory) ]
   EXEC: [ db/upgrade.sh ]
   MESSAGE: [ Upgrading database ]
   ID: [ Upgrade DB ]
Node [job] with parent [(nil)]
   EXEC: [ cp -Rv $(artifacts) $(target_directory) ]
   MESSAGE: [ Copying files to $(target_directory) ]
   ID: [ Copy files ]
Node [job] with parent [(nil)]
   HASH: [ $(exec git log -1 | head -n 1 | cut -f 1 -d \  ) ]
   MESSAGE: [ Setting finished variables ]
   ID: [ Done ]
